subworkflows/local/prepare_header.nf

```diff
@@ -0,0 +1,39 @@
+// Create a SAM header template from NCBI assembly report and SAMtools .dict
+include { BUILD_SAM_HEADER } from '../../modules/local/build_sam_header'
+
+workflow PREPARE_HEADER {
+
+    take:
+    dict    // file: /path/to/genome.dict
+    report  // file: /path/to/genome.assembly_report.txt
+    source  // file: /path/to/SOURCE (ftp path as string)
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Normalize the dict ID by removing .masked.ncbi if present and join
+    dict_mapped = dict.map { meta, path ->
+        def id = meta.id.replace('.masked.ncbi', '')
+        [id, [meta, path]]
+    }
+    report_mapped = report.map { meta, path -> [meta.id, path] }
+    source_mapped = source.map { meta, path -> [meta.id, path] }
+
+    joined = dict_mapped.join(report_mapped).join(source_mapped)
+
+    // Create input tuple with original meta.id
+    formatted_joined = joined.map { id, dict_tuple, report_path, source_path ->
+        def original_meta = dict_tuple[0] // Get the original ID with .masked.ncbi if present
+        def dict_path = dict_tuple[1]
+        return [original_meta, dict_path, report_path, source_path]
+    }
```

Comment:
I wonder if you could simplify this further by flattening `dict_mapped` above:
```
    dict_mapped = dict.map { meta, path -> [id, meta, path] }
```

```suggestion
    // Create input tuple with original meta.id
    formatted_joined = joined.map { id, original_meta, dict_path, report_path, source_path ->
        return [original_meta, dict_path, report_path, source_path]
    }
```

Reply:
Can probably reformat the whole thing like this
```
    formatted_joined = dict_mapped
                       | join(report_mapped)
                       | join(source_mapped)
                       | map { it[1..-1] }  // To remove the leading meta.id
```
although `it[1..-1]` is perhaps not the most readable / understand thing.

Reply:
👍 I like that better